captureData.js

```diff
@@ -0,0 +1,47 @@
+function captureData() {
+
+  browser.webRequest.onResponseStarted.addListener(collectRequests,
+        {urls: ['<all_urls>']}
+  );
+
+  async function collectRequests(response) {
+    const tab = await browser.tabs.get(response.tabId);
+    let tabUrl = new URL(tab.url);
+    tabUrl = removeSubdomain(tabUrl.hostname);
+    let targetUrl = new URL(response.url);
+    targetUrl = removeSubdomain(targetUrl.hostname);
+    let originUrl = new URL(response.originUrl);
+    originUrl = removeSubdomain(originUrl.hostname);
+
+    // capture first party requests
+    const firstPartyData = {
+      faviconUrl: tab.favIconUrl,
+      // access determined by storage
+      firstAccess: 0,
+      lastAccess: 0
+    };
+    console.log('storage.firstPartyRequest:', tabUrl, firstPartyData);
+
+    // capture third party requests
+    if (targetUrl !== originUrl) {
+      const thirdPartyData = {
+        document: tabUrl,
+        target: targetUrl,
+        origin: originUrl,
+        requestTime: response.timeStamp
+      };
+      console.log('storage.thirdPartyRequest:', tabUrl, thirdPartyData);
+    }
+  }
+
+  // Any subdomains can cause redundancy in storage. Returns the URL with no subdomains.
+  function removeSubdomain(url) {
```

Comment:
This also won't account for Left to Right domains... it's a nasty problem we likely won't want to solve.